notebooks/en/semantic_cache_chroma_vector_database.ipynb

```diff
@@ -1,48 +1,10 @@
 {
```

Comment:
<p>"I have included First In First Out (FIFO) eviction policy in the <code>SemanticCache</code> class to improve efficiency and flexibility. By introducing eviction policies, we provide users with the ability to control how the cache behaves when it reaches maximum capacity. This is crucial for maintaining optimal cache performance and for handling situations where memory is constrained.</p><p> </p><p>Looking at the structure of the cache, the FIFO eviction policy is straightforward. Whenever a new question-answer pair is added to the cache, it's appended to the end of the lists. Thus, the oldest (first-in) items are at the beginning of the lists. When the cache reaches its maximum size and you need to evict an item, you remove (pop) the first item from each list.</p><p> </p><p>However, the Least Recently Used (LRU) eviction policy is more complex because it requires knowledge of when each item in the cache was last accessed."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/99/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>